app/ide-desktop/lib/dashboard/src/authentication/src/authentication/cognito.ts

```diff
@@ -0,0 +1,129 @@
+/** @file Provides {@link Cognito} class which is the entrypoint into the AWS Amplify library.
+ *
+ * All of the functions used for authentication are provided by the AWS Amplify library, but we
+ * provide a thin wrapper around them to make them easier to use. Mainly, we perform some error
+ * handling and conditional logic to vary behaviour between desktop & cloud. */
+import * as amplify from '@aws-amplify/auth'
+import * as cognito from 'amazon-cognito-identity-js'
+import * as results from 'ts-results'
+
+import * as config from './config'
+
+
+
+// ===============
+// === Cognito ===
+// ===============
+
+export interface Cognito {
+    /** Returns the current user's session.
+     *
+     * Will refresh the session if it has expired.
+     *
+     * @returns `UserSession` if the user is logged in, `None` otherwise.
+     * @throws An error if failed due to an unknown error. */
+    userSession: () => Promise<results.Option<UserSession>>
+}
+
+
+
+// ===================
+// === CognitoImpl ===
+// ===================
+
+export class CognitoImpl implements Cognito {
+    constructor(
+        amplifyConfig: config.AmplifyConfig
+    ) {
+        /** Amplify expects `Auth.configure` to be called before any other `Auth` methods are
+         * called. By wrapping all the `Auth` methods we care about and returning an `Cognito` API
+         * object containing them, we ensure that `Auth.configure` is called before any other `Auth`
+         * methods are called. */
+        const nestedAmplifyConfig = config.toNestedAmplifyConfig(amplifyConfig)
+        amplify.Auth.configure(nestedAmplifyConfig)
+    }
+
+    // === Interface `impl`s ===
+
+    userSession = userSession
+}
+
+
+
+// ====================
+// === AssertString ===
+// ====================
+
+/** Type signature for a function that asserts that a parameter is a string. */
+type AssertString = (param: any, message: string) => asserts param is string
+
+/** Asserts that a parameter is a string.
+ *
+ * Used both to assert that a parameter is a string at runtime, and to inform TypeScript that a
+ * parameter is a string.
+ *
+ * @param param - The parameter to assert.
+ * @param message - The error message to throw if the assertion fails.
+ * @throws An error if the assertion fails. */
```

Comment:
FYI, we do NOT require writing `@param` nor `@returns` in functions. Use it only if you feel it really makes sense - when the argument names are not good enough to understand the context. We disabled this lint in order to make development faster.

Also, I think these `@param` docs should be removed. For example `@param param - The parameter to assert.` does not tell me anything new. The same applies to `@param message - The error message to throw if the assertion fails.`. Please check similar places and remove docs that are unneeded to make the code shorter.